lib/ansible/module_utils/vmware.py

```diff
@@ -1140,10 +1142,14 @@ def get_all_host_objs(self, cluster_name=None, esxi_host_name=None):
                 else:
                     self.module.fail_json(changed=False, msg="Cluster '%s' not found" % cluster_name)
             elif esxi_host_name:
-                esxi_host_obj = self.find_hostsystem_by_name(host_name=esxi_host_name)
-                if esxi_host_obj:
-                    host_obj_list = [esxi_host_obj]
-                else:
-                    self.module.fail_json(changed=False, msg="ESXi '%s' not found" % esxi_host_name)
+                if not isinstance(esxi_host_name, list):
+                    esxi_host_name = [esxi_host_name]
+
+                for host in esxi_host_name:
```

Comment:
Isn't this code going to be executed even if `esxi_host_name` is not a list? Causing an exception or unexpected behaviour?

Reply:
I didn't follow you here.